packages/server/src/__tests__/model-state.test.ts

```diff
@@ -15,6 +15,26 @@ describe('model-state', () => {
     setS((s) => s + 1)
     expect(s.get()).toBe(1)
   })
+  test('mutation setter with no update', () => {
+    const handler = vi.fn()
+    const [s, setS] = state(0)
+    const release = s.subscribe(handler)
+    expect(handler).toBeCalledTimes(1)
+    expect(handler).toHaveBeenLastCalledWith(0)
+    expect(s.get()).toBe(0)
+    setS((s) => s)
+    expect(handler).toBeCalledTimes(1)
```

Comment:
I messed up the comments in this test. THIS is the main thing that we are checking, that the handler does not get called again after doing `setS(s => s)`